src/sw_interface/BaseControllerROS.cpp

```diff
@@ -440,20 +446,86 @@ void BaseControllerROS::publishLiftPos()
    std::cout << std::endl;
 }
 
+void BaseControllerROS::cbCmdVel(const geometry_msgs::Twist::ConstPtr& cmd_vel)
+{
+   _stamp_cmd_vel     = ros::Time::now();
+   _cmd_vel._x_ms     = cmd_vel->linear.x;
+   _cmd_vel._y_ms     = cmd_vel->linear.y;
+   _cmd_vel._yaw_rads = cmd_vel->angular.z;
+}
+
+bool BaseControllerROS::resetOdometry(evo_rd_platform_example::resetOdomRequest& req, evo_rd_platform_example::resetOdomResponse& res)
+{
+   evo::log::get() << _logger_prefix << "Resetting Odometry.." << evo::info;
+
+   // stop and disable motors to prevent reset mid drive
+   _mecanum_drive.setCmdVel(MecanumVel());
+   _motor_handler.disableAllDriveMotors();
+   if(!_mecanum_drive.resetEncoders())
+   {
+       evo::log::get() << _logger_prefix << "Couldn't reset encoders!" <<  evo::error;
+       return false;
+   }
+   _motor_handler.enableAllDriveMotors();
+   
+   _odom_pose.reset();
+   return true;
+}
+
 void BaseControllerROS::cbCmdLift(const std_msgs::Int8::ConstPtr& cmd_lift)
 {
    _stamp_cmd_lift = ros::Time::now();
    _cmd_lift       = cmd_lift->data;
 }
 
+bool BaseControllerROS::checkFirmwareVersion(const int major_ver, const int minor_ver, const int patch_ver)
+{
+   bool success = true;
+   if(EVO_MBED_TOOLS_VER_MAJOR != major_ver)
+   {
+      evo::log::get() << _logger_prefix << "Major Version mismatch! Expected [" 
+                      << major_ver << "] but evo_mbed is [" << EVO_MBED_TOOLS_VER_MAJOR << "]!" <<  evo::error;
+      success &= false;
+   }
+
+   if(EVO_MBED_TOOLS_VER_MINOR != minor_ver)
+   {
+      evo::log::get() << _logger_prefix << "Minor Version mismatch! Expected [" 
+                      << minor_ver << "] but evo_mbed is [" << EVO_MBED_TOOLS_VER_MINOR << "]!" <<  evo::error;
+      success &= false;
+   }
+
+   // patch is not critical, so dont end here
+   if(EVO_MBED_TOOLS_VER_PATCH < patch_ver)
+   {
+      evo::log::get() << _logger_prefix << "Detected Patch Version mismatch! Expected at least [" 
+                      << patch_ver << "] but evo_mbed is [" << EVO_MBED_TOOLS_VER_MINOR << "]!" <<  evo::warn;
+   }
```

Comment:
MAJOR Version: Compability to the previous version is not given. Check is ok.
MINOR Version: Backwards compability to previous version has to be ensured. So please chance the check to (EVO_MBED_TOOLS_VER_MINOR < minor_ver)

Reply:
fixed this with commit   048a2e7

Reply:
No value is returned from bool BaseControllerROS::checkMbedLibVersion()

"return success;" after patch level Test.